### ISSUE TITLE
Incorrect callback behaviour when "Google Fonts" requests fail

### ISSUE THREAD

**Issue by anasonline** [SourcePointUSA/android-cmp-app#258]
The consent layer's Webview does apparently use Google Fonts, thus it makes two Google Fonts requests to:

`fonts.gstatic.com` and `fonts.googleapis.com`

The problem is that if one of those requests fails (e.g. connection drops or 403 server error) then the SDK responds with what looks like an incorrect callback chain which looks like the following:

- If `fonts.gstatic.com` fails -> the SDK responds with:

`OnConsentUIReady -> onConsentUIFinished -> OnError -> OnConsentUIFinished -> OnError -> OnConsentUIFinished -> OnError`

- If `fonts.googleapis.com` fails -> the SDK responds with:

`OnConsentUIReady-> OnConsentUIFinished -> OnError`

In both cases, `OnError` comes too late while it should be the first and only returned callback, and as a result to OnConsentUIReady and onConsentUIFinished getting called successively, it means that when the app is installed for the first time, the consent layer appears for a second before it gets immediately dismissed. 

Account Id: 75
Property Id: 5594
Property Name: BILD-NEWS-ANDROID

** About your environment **

SDK version: 5.3.6
Device: Google Pixel 3a
OS: Android 10

**To Reproduce**
1. Use Charles proxy or a similar application to block calls to one of the mentioned URLs
2. Install and start the app 
3. The user will see the consent layer appears and disappears quickly

**Expected behavior**
The SDK should return an onError callback which we can use to handle this error case on the client-side. The consent layer should not be displayed at all in that case.

**Comment by andresilveirah** [comment]
@anasonline thank you for bringing this up. I believe this is a duplicate of the issue you just opened #257 , the difference being the service/endpoint/url blocked by the proxy tool. We'll discuss with our frontend team the possibility of addressing this unexpected behaviour for all assets/urls so I'll close this issue in favour of #257

**Comment by anasonline** [comment]
Hi @andresilveirah and thank you for your quick response. We are also facing similar issues when the following request fails:

`https://d2p3zdq8vjvnxd.cloudfront.net/?s_id=5593&m_id=134100&consentUUID=8045a218-77f5-40d7-95e2-9504f0a9379d&requestUUID=undefined`

Do I need to open a ticket for that, or are you going to also consider this as part of the same fix needed for https://github.com/SourcePointUSA/android-cmp-app/issues/257 ?

Thanks

**Comment by andresilveirah** [comment]
@anasonline ideally we should inspect each and every network call and come up with a more complete solution for #257
